src/equations/compressible_euler_3d.jl

```diff
@@ -323,6 +323,52 @@ Details about the 1D pressure Riemann solution can be found in Section 6.3.3 of
                  zero(eltype(u_inner))) * norm_
 end
 
+"""
+    boundary_condition_slip_wall(u_inner, orientation, direction, x, t,
+                                 surface_flux_function, equations::CompressibleEulerEquations3D)
+
+Should be used together with [`TreeMesh`](@ref).
+"""
+@inline function boundary_condition_slip_wall(u_inner, orientation,
+                                              direction, x, t,
+                                              surface_flux_function,
+                                              equations::CompressibleEulerEquations3D)
+  # get the appropriate normal vector from the orientation
+  if orientation == 1
+    normal_direction = SVector(1.0, 0.0, 0.0)
+  elseif orientation == 2
+    normal_direction = SVector(0.0, 1.0, 0.0)
+  else # orientation == 3
+    normal_direction = SVector(0.0, 0.0, 1.0)
+  end
+
+  # compute and return the flux using `boundary_condition_slip_wall` routine above
+  return boundary_condition_slip_wall(u_inner, normal_direction, direction,
+                                      x, t, surface_flux_function, equations)
+end
+
+"""
+    boundary_condition_slip_wall(u_inner, normal_direction, direction, x, t,
+                                 surface_flux_function, equations::CompressibleEulerEquations3D)
+
+Should be used together with [`StructuredMesh`](@ref).
```

Comment:
I don't think it fully works with curved `DGMultiMesh`es yet, but this is because the parabolic terms are assuming geometric terms are constant over each element. Once that's addressed, then yes, this would work for curved `DGMultiMesh` too.

Reply:
Because this routine is for compressible Euler slip walls it probably works for `P4estMesh` is that context in 3D (I did not test if this is true). However, as @jlchan said, the parabolic terms are only implemented for constant metric terms at the moment. So for mixed hyperbolic/parabolic runs this boundary condition will only work with `TreeMesh` at the moment.

Reply:
OK, thanks!